src/main/kotlin/racingcar/interfaces/InputAdapter.kt

```diff
@@ -0,0 +1,63 @@
+package racingcar.interfaces
+
+import racingcar.domain.Car
+
+class InputAdapter(
+  private val carNames: String,
+  private val matches: String
+) {
+  fun splitCarName(): List<String> {
+    val cars = carNames.split(DELIMITER)
+    return cars
+  }
+
+  fun validateCarName(cars: List<String>): Set<String> {
+    val racingCars = mutableSetOf<String>()
+    for (car in cars) {
+      when {
+        car.length > 5 -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_LESS_THAN_5_EXCEPTION_MSG)
+        car.isEmpty() -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_EMPTY_EXCEPTION_MSG)
+        car.isBlank() -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_WHITESPACE_EXCEPTION_MSG)
+        racingCars.contains(car) -> throw IllegalArgumentException(INPUT_ERROR_MSG + CARNAME_SAME_CARNAME_EXCEPTION_MSG)
+      }
```

Comment:
```suggestion
      require(car.length <= 5) { INPUT_ERROR_MSG + CARNAME_LESS_THAN_5_EXCEPTION_MSG }
      require(car.isNotEmpty()) { INPUT_ERROR_MSG + CARNAME_EMPTY_EXCEPTION_MSG }
      require(car.isNotBlank()) { INPUT_ERROR_MSG + CARNAME_WHITESPACE_EXCEPTION_MSG }
      require(car !in racingCars) { INPUT_ERROR_MSG + CARNAME_SAME_CARNAME_EXCEPTION_MSG }
```

이번 과제를 진행하게 되면서 코틀린의 require() 함수를 알게 되었습니다.
기존의 코드도 좋지만 require()는 인자의 조건이 false일 때, 중괄호 안의 메시지로 IllegalArgumentException을 발생시키며 코틀린에서 권장하는 방법이라고 합니다. 이 경우에는 when 블록도 제거할 수 있어서 더욱 효과적일 것 같습니다.

그리고 코틀린에선 가독성을 위해 `contains()`보다 `in` 문법을 사용하는것을 권장한다고 하며 정확히 같은 의미라고 하네요

Reply:
감사합니다!

Reply:
이번주에 require()부분을 공부해봐야겠어요! 감사합니다!